### PR TITLE
Add support for homebrew packages

### PR DESCRIPTION
**Pull request by tushar912**
I have added support for homebrew packages.

### CODE REVIEW

**Comment by TG1999** [comment]
Please use "black" for formatting

**Comment by tushar912** [comment]
On the entire code?.... I thought that it might disturb the rest of code.

**Comment by TG1999** [comment]
Run black on the file that you are submitting for review

**Comment by tushar912** [comment]
Ok

**Comment by tushar912** [comment]
@TG1999 I have ran black on the files I changed.

File: tests/test_package.py [hunk 1]
```diff
@@ -97,3 +97,13 @@ def test_rubygems_packages(mock_get):
     mock_get.side_effect = side_effect
     packages = list(info(purl))
     match_data(packages, expected_data)
+
+
+@mock.patch("fetchcode.package.get_response")
+def test_homebrew_packages(mock_get):
+    side_effect = [file_data("tests/data/homebrew_mock_data.json")]
+    purl = "pkg:homebrew/node"
```
Comment (TG1999) [inline tests/test_package.py:105]: Please make one more test case where purl has a version in it.

**Comment by tushar912** [comment]
@TG1999 I have added additional test case.

**Comment by TG1999** [comment]
Hey @tushar912 please check what a version inside purl means, checkout packageurl-python to know more

**Comment by tushar912** [comment]
Oh I misunderstood , I would make the change ASAP.

**Comment by TG1999** [comment]
@pombredanne @majurg ping

**Review by TG1999 — APPROVED** [review]

File: fetchcode/package.py [hunk 2]
```diff
@@ -327,3 +327,36 @@ def get_rubygems_data_from_purl(purl):
...
+    )
+    versions = response.get("versions") or {}
+    version_purl = PackageURL(type=purl.type, name=name, version=versions.get("stable"))
+    yield Package(
+        homepage_url=homepage_url,
+        api_url=api_url,
+        declared_license=declared_license,
+        download_url=download_url,
+        **version_purl.to_dict(),
+    )
```
Comment (majurg) [inline fetchcode/package.py:362]: If there is more than one entry in `versions`, wouldnt we need to loop thru them all?

Comment (pombredanne) [reply fetchcode/package.py:362]: @majurg I checked and from what I can see there is only one version of interest under the attribute "stable"  
@tushar912 do you concur?

File: fetchcode/package.py [hunk 3]
```diff
@@ -327,3 +327,36 @@ def get_rubygems_data_from_purl(purl):
...
+    name = purl.name
+    base_path = "https://formulae.brew.sh/api/formula"
+    api_url = f"{base_path}/{name}.json"
+    response = get_response(api_url)
+    declared_license = response.get("license")
+    homepage_url = response.get("homepage")
+    urls = response.get("urls") or {}
+    stable_url = urls.get("stable") or {}
+    download_url = stable_url.get("url")
+    yield Package(
```
Comment (pombredanne) [inline fetchcode/package.py:347]: Why do we yield a package without a version and one with a version?

**Comment by pombredanne** [comment]
This is highly similar (of course) to the code in https://github.com/nexB/fetchcode/pull/44 .... when merging (it still needs love) I think it would be best to credit both authors.

**Comment by pombredanne** [comment]
@tushar912 gentle ping ... do you still want to finish merge this so we can merge? If not we may close it soon. Thanks!

**Comment by keshav-space** [comment]
@tushar912 closing this, feel free to reopen if you’re still interested in getting this merged.

**Pull request closed without merging**